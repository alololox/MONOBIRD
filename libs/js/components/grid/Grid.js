import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

const Wrapper = styled.main`
  display: flex;
  flex-flow: column wrap;
  padding: 0 1rem;
  width: 88%;
  margin: 0 auto;
  max-width: 140rem;
`;

const Row = styled.div`
  display: flex;
  margin-bottom: 2rem;
`;

const Item = styled.div`
  display: flex;
  flex-grow: 1;
  background-color: grey;
`;

const Grid = () => (
  <Wrapper>
    <Row>
      <Item>1</Item>
      <Item>2</Item>
      <Item>3</Item>
    </Row>

    <Row>
      <Item>4</Item>
      <Item>5</Item>
      <Item>6</Item>
    </Row>
  </Wrapper>
);

export default Grid;
