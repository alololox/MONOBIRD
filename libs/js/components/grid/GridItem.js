import React from "react";

const GridItem = () => <main>!</main>;

export default GridItem;
