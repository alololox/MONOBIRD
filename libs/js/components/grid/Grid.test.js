import React from "react";
import { render } from "@testing-library/react";

import Grid from "./Grid";
import GridItem from "./GridItem";

import "jest-styled-components";

describe("Dirty Little Grid Test", () => {
  it("Renders Grid", () => {
    const { asFragment } = render(<Grid />);

    expect(asFragment()).toMatchSnapshot();
  });
});
