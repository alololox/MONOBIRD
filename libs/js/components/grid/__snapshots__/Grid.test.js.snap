// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dirty Little Grid Test Renders Grid 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column wrap;
  -ms-flex-flow: column wrap;
  flex-flow: column wrap;
  padding: 0 1rem;
  width: 88%;
  margin: 0 auto;
  max-width: 140rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 2rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  background-color: grey;
}

<main
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        1
      </div>
      <div
        class="c2"
      >
        2
      </div>
      <div
        class="c2"
      >
        3
      </div>
    </div>
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        4
      </div>
      <div
        class="c2"
      >
        5
      </div>
      <div
        class="c2"
      >
        6
      </div>
    </div>
  </main>
</DocumentFragment>
`;
